packages/cosmos_ui_components/lib/components/cosmos_text_field.dart

```diff
@@ -0,0 +1,60 @@
+import 'package:cosmos_ui_components/cosmos_ui_components.dart';
+import 'package:flutter/material.dart';
+
+class CosmosTextField extends StatefulWidget {
+  final String text;
+  final Function(String) onChanged;
+  final int? maxLines;
+  final int maxLength;
+
+  const CosmosTextField({
+    Key? key,
+    required this.onChanged,
+    this.maxLength = 50,
+    this.text = '',
+    this.maxLines,
+  }) : super(key: key);
+
+  @override
+  State<CosmosTextField> createState() => _CosmosTextFieldState();
+}
+
+class _CosmosTextFieldState extends State<CosmosTextField> {
+  final TextEditingController controller = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
+    controller.text = widget.text;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = CosmosTheme.of(context);
+    return TextField(
+      controller: controller,
+      maxLines: widget.maxLines,
+      maxLength: widget.maxLength,
+      onChanged: widget.onChanged,
+      decoration: InputDecoration(
+        border: const UnderlineInputBorder(),
+        counterText: '',
+        suffix: SizedBox(
+          height: 17,
+          width: 17,
+          child: CircleAvatar(
+            backgroundColor: theme.colors.divider,
+            foregroundColor: theme.colors.background,
+            child: InkWell(
+              onTap: () {
+                controller.clear();
+                widget.onChanged('');
+              },
+              child: const Icon(Icons.clear, size: 12),
```

Comment:
we should refrain from using default flutter icons, as those are subject to change and differ between platforms. We should always pull the icons from designs and place them in assets, take a look into `CosmosBackButton` and `"assets/images/big-arrow-left.png"` for example

Reply:
Updated